src/ngMock/angular-mocks.js

```diff
@@ -1341,13 +1341,13 @@ function createHttpBackendMock($rootScope, $timeout, $delegate, $browser) {
       // service the ngMockE2E version can pass through to the real backend
       originalHttpBackend = $delegate.$$originalHttpBackend || $delegate;
 
-  function createResponse(status, data, headers, statusText) {
+  function createResponse(status, data, headers, statusText, xhrStatus) {
     if (angular.isFunction(status)) return status;
 
     return function() {
       return angular.isNumber(status)
-          ? [status, data, headers, statusText]
-          : [200, status, data, headers];
+          ? [status, data, headers, statusText, xhrStatus || (status === 200 ? 'success' : 'error')]
```

Comment:
Since `xhrStatus` is something that is generated by `$httpBackend` (i.e. it is not something that comes from the server, I think we should not ven make it "trainable". I.e. you should:

1. Remove the `xhrStatus` argument from `createResponse`  (and other similar places).
2. Generate `xhrStatus` inside `createResponse`, based on the same rules we use in `$httpBackend` (i.e. `200 <= status < 300 ? 'success' : 'error'`).
3. Update existing tests. Add new tests that verify this new behavior.